crates/astria-sequencer/src/grpc/sequencer.rs

```diff
@@ -118,14 +118,21 @@ impl SequencerService for SequencerServer {
                 ))
             })?;
 
-        let all_rollup_ids = snapshot
+        let mut all_rollup_ids = snapshot
             .get_rollup_ids_by_block_hash(&block_hash)
             .await
             .map_err(|e| Status::internal(format!("failed to get rollup ids from storage: {e}")))?
             .into_iter()
             .map(RollupId::to_vec)
             .collect::<Vec<_>>();
+        all_rollup_ids.sort_unstable();
 
+        // Filter out the Rollup Ids requested which have no data before grabbing
+        // so as to not error because the block had no data for the requested rollup
+        let rollup_ids: Vec<RollupId> = rollup_ids
+            .into_iter()
+            .filter(|id| all_rollup_ids.binary_search(&id.to_vec()).is_ok())
```

Comment:
I prefer to not turn all rollup IDs into vecs prematurely, but even so this allocation could still be avoided by using [`slice::binary_search_by`](https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search_by): `.filter(|id| all_rollup_ids.binay_search_by(|probe| probe.as_slice().cmp(id.as_ref()))`